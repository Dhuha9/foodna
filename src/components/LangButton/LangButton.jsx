import React from 'react'
import { Form } from 'react-bootstrap';
import 'bootstrap/dist/css/bootstrap.min.css';
import { useTranslation} from 'react-i18next';

export default function LangButton() {

    const { i18n } = useTranslation();

    const changeLanguage = (lng) => {
      i18n.changeLanguage(lng);
    };
  
    const currentLang = i18n.language || window.localStorage.i18nextLng || '';
  
    return (
      <div >
        <Form>
          <Form.Group >
              <Form.Control 
                as="select" 
                size="sm" 
                value={currentLang} onChange={() => changeLanguage(event.target.value)}
                style={{width: `90px`}}
                >
                  <option value="en">English</option>
                  <option value="ar">Arabic</option>
              </Form.Control>
          </Form.Group>
        </Form>
      </div>
    );
}
