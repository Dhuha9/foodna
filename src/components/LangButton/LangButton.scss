.langButton {
  box-shadow: 0 1px 2px 0 rgba(0, 0, 0, 0.2);
  border-radius: 2px;
  border: 1px solid transparent;
  width: 95px;
}
