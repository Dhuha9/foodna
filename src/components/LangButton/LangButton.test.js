import React from 'react';
import renderer from 'react-test-renderer';

import LangButton from './LangButton';

it ("renders when there is no LangButton", () => {
    const tree = renderer.create(
        <LangButton />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});