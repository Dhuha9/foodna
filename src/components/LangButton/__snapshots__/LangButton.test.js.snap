// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders when there is no LangButton 1`] = `
<div>
  <form
    className=""
  >
    <div
      className="form-group"
    >
      <select
        className="form-control form-control-sm"
        onChange={[Function]}
        style={
          Object {
            "width": "90px",
          }
        }
        value=""
      >
        <option
          value="en"
        >
          English
        </option>
        <option
          value="ar"
        >
          Arabic
        </option>
      </select>
    </div>
  </form>
</div>
`;
